### DISCUSSION TITLE
Please add "class word" and "token" options!

### DISCUSSION THREAD

**Discussion by Kvento** [d8ahazard/sd_dreambooth_extension#248] [Ideas]
It would be super cool to have the same options for training as Dreambooth from Joe Penna. 
Because right now we have **"instance prompt"** and **"class prompt"**, but Joe Penna's version there are **"class word"** and **"token"** options and it's **much easier** to use to training your own style and to get good result.

![4145614](https://user-images.githubusercontent.com/15001752/202560906-0cc0c925-be63-4b19-8e7c-f6b3f5419905.png)

Because you only need to enter this two words (**style** to say - I want to train the style, and your style's **unique name**) and I noticed that it uses a random prompt for training the style.

![45154155](https://user-images.githubusercontent.com/15001752/202562398-a7f2f6cc-880b-49eb-98a2-86418a665927.png)

![141455](https://user-images.githubusercontent.com/15001752/202562426-0513e1a4-1af8-412d-a4e6-71417d5c3953.png)

👍🏻 👍🏻 👍🏻 👍🏻 👍🏻 

### **Can it be implemented in this addon?** Please, please, please!

👍🏻 👍🏻 👍🏻 👍🏻 👍🏻

**Comment by d8ahazard** [comment]
What are the randomized examples actually used for? From the code sample you posted, that appears to be just what's used to generate the "sample" image.

Just from what you've written, I think we already have these features, aside from the sample prompt stuff, and even that is just "kind of".

I think the token is just the instance prompt, and class word is...class prompt. Both should work the same, with the exception of with ours, you can caption all of your images "a painting in the style of style" and then use "style" for your class prompt, and "[filewords] mysytle" for the instance prompt...or just "[filewords]" for your class prompt. This will read the captions from each image, and replace [filewords] with the caption to create the prompts. 

Maybe I'm mistaken, but it looks like that's what his code is doing.

If you want me to add a list of random examples for the sample prompts, that I can probably do.

**Reply by Kvento** [reply]
`If you want me to add a list of random examples for the sample prompts, that I can probably do.`

Yes, that would be great!

`you can caption all of your images "a painting in the style of style" and then use "style" for your class prompt, and "[filewords] mysytle" for the instance prompt...or just "[filewords]" for your class prompt. This will read the captions from each image, and replace [filewords] with the caption to create the prompts.`

Ok, I'll try, thanks for advice!
